### PR TITLE
Refresh retained released/not_active metadata on deferred blocker refetch

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- refresh deferred retained `released/not_active` blocker metadata after fresh-discovery refetches
- route terminal same-issue truth through the normal release path and keep single-claim refresh failures from aborting the rest of fresh discovery
- add focused regressions and refresh the CO-292 task packet after the clean review rerun and elegance pass

## Testing
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npx vitest run orchestrator/tests/ProviderIssueHandoff.test.ts -t "refetches steady-state blocker snapshots before refreshing retained released metadata|releases terminal retained metadata after a fresh-discovery blocker refetch|keeps fresh discovery running when a deferred retained blocker refresh fails"`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run pack:smoke`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`

Diff budget override: CO-292 needs the paired provider handoff fix, focused regression coverage, and issue-scoped docs updates together; splitting further would break the required audit trail and regression proof.

## Summary by CodeRabbit

* **New Features**
  * Automatic metadata refresh for retained claims in released/not-active status: cached issue state, state type and update timestamp are kept aligned with newer live issue information while preserving the claim’s released status.

* **Documentation**
  * Added PRD, technical spec, action plan, task checklist and registry entries describing scope, acceptance criteria, validation steps and handoff status.

* **Tests**
  * Extensive new test coverage validating retained-not-active metadata refresh behaviours and edge cases.

### CODE REVIEW

File: orchestrator/src/cli/control/providerIssueHandoff.ts [hunk 1]
```diff
@@ -5389,6 +5519,150 @@ export function createProviderIssueHandoffService(
...
+                  }
+                );
+                if (
+                  canRefreshRetainedReleasedNotActiveClaimMetadataOnly({
+                    claim,
+                    trackedIssue: directResolution.trackedIssue
+                  }) &&
+                  (eligibility.eligible ||
+                    eligibility.releaseReason === 'provider_issue_released:not_active')
+                ) {
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerIssueHandoff.ts:5596]: Intentional for CO-292. The issue spec keeps `Blocked -> Rework` as a metadata-only refresh on the retained row, not a reopen path: `docs/PRD-linear-d2306cf9-2c2f-4242-bbaa-2e86456221a6.md` says the row remains `released/not_active`, and the focused regression `refreshes retained released not-active metadata from dependent blocker truth without reclaiming` locks that behavior in. Ready reclaim/admission stays out of scope here and is covered by the separate CO-202 / CO-212 family, so reopening from this deferred refresh helper would regress the intended boundary.
